docs/qute/TemplateSupport.md

```diff
@@ -0,0 +1,93 @@
+
+## Parameter declaration support
+
+A `parameter declaration` is used to declare the expected model in a Qute template:
+
+```
+{@java.util.List<org.acme.qute.Item> items}
+```
+
+The Qute support helps to easily write this declaration with the completion, hover and definition support.
+
+### Completion for Java types
+
+The Qute support provides completion to easily write parameter declaration:
+
+![Completion for Java types](images/QuteTemplateParameterDeclarationCompletion.gif)
+
+### Validate Java type
+
+The Qute support validates Java types:
+
+![Validate Java type](images/QuteTemplateParameterDeclarationValidation.png)
+
+### Jump to Java type definition
+
+You can jump to the proper Java type:
+
+![Jump to Java type definition](images/QuteTemplateParameterDeclarationDefinition.gif)
+
+## Section support
+
+### Completion for Qute section
+
+Qute support provides snippets to quickly generate `Qute` sections. Here is a sample for the `#for` section:
+
+![Completion for Qute section](images/QuteTemplateSectionSnippetCompletion.gif)
+
+### Linked editing
+
+Linked editing is supported, allowing for simultaneous renaming of a parameter expression.
+
+To enable this feature, the setting `editor.linkedEditing` must be set to `true` in your `settings.json`.
+
+![Linked EditingRange](images/QuteTemplateLinkedEditingRange.gif)
+
+## Tag Name Highlighting
+
+When placing the cursor on a Qute section tag, the corresponding opening/closing tag will be highlighted with their nested block:
+
+![Highlight Corresponding Tag](images/QuteTemplateSectionHighlighting.gif)
+
+## Expression support
+
+A Qute expression is declared inside `{|}` or inside a section (ex : `{#for item in |}`).
+
+### Completion for Java data model
+
+Completion is available for allowed Java `fields`, `methods` and `virtual methods` (coming from the Qute core like `orEmpty` or coming from @TemplateExtension):
+
+![Completion for Java data model](images/QuteTemplateCompletionForJavaDataModel.gif)
+
+### Validation for Java data model
+
+Validation highlights unknown Java fields or methods as an error:
+
+![Validation for Java data model](images/QuteTemplateValidationForJavaDataModel.png)
+
+### Hover for Java data model
+
+When you hover over an object, property or method, you can see the Java type:
+
+![Hover for Java data model](images/QuteTemplateCompletionForJavaDataModel.gif)
+
+### Jump to Java field, method definition
+
+`Ctrl+Click` on a property or method to jump to the definition of the matching Java field, method or virtual method (for `@TemplateExtension`):
+
+![Definition for Java data model](images/QuteTemplateDefinitionForJavaDataModel.gif)
+
+## Quarkus integration for Qute
+
+When the template is used in a Quarkus application and it is bound with a Java Template field or a method annotated with `@CheckedTemplate`, you can see the expected model as CodeLens:
+
+![CodeLens for Java data model](images/QuteTemplateCodeLensJavaDataModel.png)

```

Comment:
Maybe clean this image up a bit and use straight lines/arrows rather then freehand

Reply:
fixed